base_layer/common_types/src/types/mod.rs

```diff
@@ -78,3 +78,18 @@ pub type RangeProofService = BulletproofsPlusService;
 
 /// Specify the range proof
 pub type RangeProof = BulletRangeProof;
+
+use tari_crypto::{hash_domain, hashing::DomainSeparatedHasher};
+
+pub(crate) const BULLET_RANGEPROOFS_HASHER_LABEL: &str = "bullet_rangeproofs.hasher";
+
+hash_domain!(
+    BaseLayerCommonTypesDomain,
+    "com.tari.tari-project.base_layer.common_types"
+);
+
+pub(crate) fn base_layer_common_types_domain_hasher(
+    label: &'static str,
+) -> DomainSeparatedHasher<Blake256, BaseLayerCommonTypesDomain> {
+    DomainSeparatedHasher::<Blake256, BaseLayerCommonTypesDomain>::new_with_label(label)
+}
```

Comment:
I think we can simplify this:
```suggestion
hash_domain!(
    BulletRangeproofsHashDomain,
    "com.tari.tari-project.base_layer.common_types.bullet_rangeproofs",
    1
);
pub type BulletRangeproofsHasherBlake256 = DomainSeparatedHasher<Blake256, BulletRangeproofsHashDomain>;
```

Reply:
Very cool ! Thanks for the suggestion